registry/txt.go

```diff
@@ -224,18 +248,31 @@ func (im *TXTRegistry) generateTXTRecord(r *endpoint.Endpoint) []*endpoint.Endpo
 			endpoints = append(endpoints, txt)
 		}
 	}
-	// new TXT record format (containing record type)
+
 	recordType := r.RecordType
 	// AWS Alias records are encoded as type "cname"
 	if isAlias, found := r.GetProviderSpecificProperty("alias"); found && isAlias == "true" && recordType == endpoint.RecordTypeA {
 		recordType = endpoint.RecordTypeCNAME
 	}
-	txtNew := endpoint.NewEndpoint(im.mapper.toNewTXTName(r.DNSName, recordType), endpoint.RecordTypeTXT, r.Labels.Serialize(true, im.txtEncryptEnabled, im.txtEncryptAESKey))
-	if txtNew != nil {
-		txtNew.WithSetIdentifier(r.SetIdentifier)
-		txtNew.Labels[endpoint.OwnedRecordLabelKey] = r.DNSName
-		txtNew.ProviderSpecific = r.ProviderSpecific
-		endpoints = append(endpoints, txtNew)
+
+	// "new" (version 2) TXT record format (containing record type)
```

Comment:
better to have a reference to a doc where `"new"  (version 2)` is defined

Reply:
AFAICT there are no docs referencing this format in the `master` branch, but all I can find is this [doc](https://github.com/kubernetes-sigs/external-dns/blob/1f6340a95c0f5a96cc5888196eb767943f96bea4/docs/registry.md) that was removed some time ago.
The new term was introduced during the proposal and it's probably not the best way to describe it, as this PR adds yet another new format.
I could link that file or we could name the format differently. wdyt?

Reply:
I would say write a section in a doc about it and reference that.

Reply:
Added a reference.